storage/types.go

```diff
@@ -277,6 +282,34 @@ func (n *NodeID) String() string {
 	return r.String()
 }
 
+// AsKey returns a string identifier for this NodeID suitable for
+// short term use e.g. as a Map key. It is more efficient to use this than
+// String() as it's not constrained to return a binary string.
+func (n *NodeID) AsKey() string {
+	var b strings.Builder
+	fullBytes := n.PrefixLenBits / 8
+	bitsLeft := n.PrefixLenBits % 8
+
+	// Note: all Builder write methods are documented never to return errors.
+	// Write the length first.
+	b.WriteString(strconv.Itoa(n.PrefixLenBits))
+	b.WriteRune(':')
+	// We can do all the full bytes in one go.
+	if fullBytes > 0 {
```

Comment:
```golang
fmt.Fprintf(&b, "%d:%x", n.PrefixLenBits, n.Path[:fullBytes])
```
?

Reply:
hex.Encode looks more efficient.

Reply:
Could you try the benchmark? Although `Fprintf` has more indirections, it writes the result straight to the buffer, whereas the current code allocates an extra string in line 295. Not a clear win to me.

Reply:
Tried it locally on my machine, `fmt.Fprintf` is 50% slower than `hex.Encode`. Makes sense to me now.